fastly/erl.go

```diff
@@ -0,0 +1,342 @@
+	ERLTrace   ERLHttpMethodsEnum = "TRACE"
+	ERLPost    ERLHttpMethodsEnum = "POST"
+	ERLHead    ERLHttpMethodsEnum = "HEAD"
+	ERLDelete  ERLHttpMethodsEnum = "DELETE"
+	ERLPatch   ERLHttpMethodsEnum = "PATCH"
+	ERLOptions ERLHttpMethodsEnum = "OPTIONS"
+)
+
+// Enum for ERL
+type ERLLoggerEnum string
+
+const (
+	ERLLogAzureBlob       ERLLoggerEnum = "azureblob"
+	ERLLogBigQuery        ERLLoggerEnum = "bigquery"
+	ERLLogCloudFiles      ERLLoggerEnum = "cloudfiles"
+	ERLLogDataDog         ERLLoggerEnum = "datadog"
+	ERLLogDigitalOcean    ERLLoggerEnum = "digitalocean"
+	ERLLogElasticSearch   ERLLoggerEnum = "elasticsearch"
+	ERLLogFtp             ERLLoggerEnum = "ftp"
+	ERLLogGcs             ERLLoggerEnum = "gcs"
+	ERLLogGoogleAnalytics ERLLoggerEnum = "googleanalytics"
+	ERLLogHeroku          ERLLoggerEnum = "heroku"
+	ERLLogHoneycomb       ERLLoggerEnum = "honeycomb"
+	ERLLogHttp            ERLLoggerEnum = "http"
+	ERLLogHttps           ERLLoggerEnum = "https"
+	ERLLogKafta           ERLLoggerEnum = "kafka"
+	ERLLogKinesis         ERLLoggerEnum = "kinesis"
+	ERLLogLogEntries      ERLLoggerEnum = "logentries"
+	ERLLogLoggly          ERLLoggerEnum = "loggly"
+	ERLLogLogShuttle      ERLLoggerEnum = "logshuttle"
+	ERLLogNewRelic        ERLLoggerEnum = "newrelic"
+	ERLLogOpenStack       ERLLoggerEnum = "openstack"
+	ERLLogPaperTrail      ERLLoggerEnum = "papertrail"
+	ERLLogPubSub          ERLLoggerEnum = "pubsub"
+	ERLLogS3              ERLLoggerEnum = "s3"
+	ERLLogScalyr          ERLLoggerEnum = "scalyr"
+	ERLLogSftp            ERLLoggerEnum = "sftp"
+	ERLLogSplunk          ERLLoggerEnum = "splunk"
+	ERLLogStackDriver     ERLLoggerEnum = "stackdriver"
+	ERLLogSumoLogiuc      ERLLoggerEnum = "sumologic"
+	ERLLogSysLog          ERLLoggerEnum = "syslog"
+)
+
+// Enum for ERL
+type ERLWindowSizeEnum int
+
+const (
+	ERLSize1  ERLWindowSizeEnum = 1
+	ERLSize10 ERLWindowSizeEnum = 10
+	ERLSize60 ERLWindowSizeEnum = 60
+)
+
+// ERLsByName is a sortable list of ERLs
+type ERLsByName []*ERL
+
+// Len, Swap, and Less implement the sortable interface
+func (s ERLsByName) Len() int      { return len(s) }
+func (s ERLsByName) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s ERLsByName) Less(i, j int) bool {
+	return s[i].Name < s[j].Name
+}
+
+// ********************************************************************************************* //
+//
+// Function 1: List rate limiters
+// Note(s):
+//
+// Return type
+//   []*ERL
+
+// Input to the ListERLs function
+type ListERLsInput struct {
+	ServiceID      string // required
+	ServiceVersion int    // required
+}
+
+// Returns the list of ERLs for the specified version
+func (c *Client) ListERLs(i *ListERLsInput) ([]*ERL, error) {
+	if i.ServiceID == "" {
+		return nil, ErrMissingServiceID
+	}
+	if i.ServiceVersion == 0 {
+		return nil, ErrMissingServiceVersion
+	}
+
+	path := fmt.Sprintf("/service/%s/version/%d/rate-limiters", i.ServiceID, i.ServiceVersion)
+	resp, err := c.Get(path, nil)
```

Comment:
I wonder, because this is implementing an LA (Limited Availability) feature, if we should (or are even able) to identify if the response is a 4xx (or 5xx)? 

Because if the user doesn't have the feature enabled on their account, it would be nice to provide extra context/remediation steps. Otherwise the user may not understand why the API call failed if they didn't notice the LA status of the feature in the documentation.

I would imagine the worst case is there would be stale code that we'd need to remove once the feature goes GA (General Availability) and so we'd want to add a TODO: ... code comment for future maintainers.

I've asked about this in the past for other LA feature sets added to the go-fastly API client and I've been told there was no such ability to distinguish errors caused by the feature not being active on the user account. So I wouldn't be surprised if the same was the case here, but wanted to raise this just in case that was possible.

Reply:
OK. Thanks!